src/utils/logger.hpp

```diff
@@ -2,23 +2,58 @@
 
 #include <fmt/printf.h>
 
-extern int (*nrnpy_pr_stdoe_callback)(int, char*);
+using LoggerCallback = int(int, char*);
 
+class Logger {
+  public:
+    void setCallback(LoggerCallback* cb) {
+        callback = cb;
+    }
+
+    LoggerCallback* getCallback() const {
+        return callback;
+    }
+
+    template <typename... Args>
+    void info(const char* fmt, Args... args) {
+        if (callback) {
+            std::string message = fmt::format(fmt, std::forward<Args>(args)...);
+            callback(1, message.data());
+        }
+        fmt::print(fmt, std::forward<Args>(args)...);
+    }
+
+    template <typename... Args>
+    void error(const char* fmt, Args... args) {
+        if (callback) {
+            std::string message = fmt::format(fmt, std::forward<Args>(args)...);
+            callback(2, message.data());
+        }
+        fmt::print(stderr, fmt, std::forward<Args>(args)...);
+    }
```

Comment:
The two methods `info` and `error` differ by a number `1` or `2`. Therefore, they can easily be deduplicated.

Reply:
Done, will be more usefull when will add the full range of output: `error`, `info`, `warning`, `debug`, ...